dio/test/http_client_adapter_test.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:dio/dio.dart';
+import 'package:dio/io.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group(HttpClientAdapter, () {
+    test(
+        'IOHttpClientAdapter.onHttpClientCreate is only executed once per request',
+        () async {
+      int onHttpClientCreateInvokeCount = 0;
+      final dio = Dio();
+      dio.httpClientAdapter = IOHttpClientAdapter(onHttpClientCreate: (client) {
+        onHttpClientCreateInvokeCount++;
+        return client;
+      });
+      await dio.get('https://pub.dev');
+      expect(onHttpClientCreateInvokeCount, 1);
```

Comment:
This can be `expectLater` then we don't need the asynchronized function here anymore.